akka-stream/src/main/scala/akka/stream/impl/Sources.scala

```diff
@@ -136,3 +139,59 @@ private[akka] final class SourceQueueAdapter[T](delegate: SourceQueue[T]) extend
   def offer(elem: T): CompletionStage[QueueOfferResult] = delegate.offer(elem).toJava
   def watchCompletion(): CompletionStage[Done] = delegate.watchCompletion().toJava
 }
+
+/**
+ * INTERNAL API
+ */
+private[stream] final class BlockingStreamSource[T, S](create: () ⇒ S,
```

Comment:
You are suggesting:
* rename blocking to resource
* add non blocking version with () => Future[S], read: S => Future[T] and S => Future[Done] 
I'm Ok with this, still would like to ask ticket author. 
@rkuhn, are you Ok to make mentioned points within this PR?